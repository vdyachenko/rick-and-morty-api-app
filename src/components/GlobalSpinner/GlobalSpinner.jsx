import React from 'react';

import { Spin } from 'antd';

import { withCenteredPosition } from 'hoc';

const GlobalSpinner = () =>  <Spin size="large" />;

export default withCenteredPosition(GlobalSpinner);