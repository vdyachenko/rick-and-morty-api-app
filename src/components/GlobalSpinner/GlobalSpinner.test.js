import React from 'react';
import { mount } from 'enzyme';

import GlobalSpinner from './GlobalSpinner';

let wrap;
describe('<GlobalSpinner />', () => {
  beforeEach(() => {
    wrap = mount(<GlobalSpinner />);
  });
  afterEach(() => {
    wrap.unmount();
  });
  it('renders without crashing', () => {
    expect(wrap).toMatchSnapshot();
  });
  it(`renders properly`, () => {
    expect(wrap.find('.ant-spin').length).toEqual(1);
    expect(wrap.find('.ant-spin').hasClass('ant-spin-lg')).toEqual(true);
  });
});