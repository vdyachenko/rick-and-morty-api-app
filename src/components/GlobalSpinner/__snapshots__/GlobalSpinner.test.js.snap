// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GlobalSpinner /> renders without crashing 1`] = `ReactWrapper {}`;
